pkg/ccl/backupccl/testdata/backup-restore/restore-regionless-alter-region

```diff
@@ -0,0 +1,341 @@
+----
+us-east-1
+
+query-sql
+SHOW TABLES FROM d;
+----
+public t table root <nil> REGIONAL BY TABLE IN PRIMARY REGION
+
+# ensure that tables belonging to d can be modified & have the correct values
+exec-sql
+INSERT INTO d.t VALUES (6), (7);
+----
+
+query-sql
+SELECT * FROM d.t;
+----
+1
+2
+3
+4
+5
+6
+7
+
+subtest end
+
+subtest end
+
+# create cluster with a single region and test db restore with strip_localities
+subtest restore_regionless_on_single_region_db
+new-cluster name=s3 share-io-dir=s1 allow-implicit-access localities=us-east-1
+----
+
+exec-sql
+RESTORE DATABASE d FROM LATEST IN 'nodelocal://1/database_backup/' WITH strip_localities;
+----
+
+# check to see if restored database, d, shows up
+query-sql
+SHOW DATABASES;
+----
+d root <nil> <nil> {} <nil>
+defaultdb root <nil> <nil> {} <nil>
+postgres root <nil> <nil> {} <nil>
+system node <nil> <nil> {} <nil>
+
+# ensure that database d is regionless
+query-sql
+SELECT region FROM [SHOW REGIONS FROM DATABASE d] ORDER BY 1;
+----
+
+# show tables - make sure these are regionless as well
+query-sql
+SHOW TABLES FROM d;
+----
+public t table root <nil> <nil>
+
+# ensure that tables belonging to d can be modified & have the correct values
+exec-sql
+INSERT INTO d.t VALUES (4), (5);
+----
+
+query-sql
+SELECT * FROM d.t ORDER BY 1;
+----
+1
+2
+3
+4
+5
+
+subtest restore_regionless_on_single_region_db/alter-region
+# ensure we can't add a regular region
+# this is due to a validation that ensures a primary region exists
+# before a "regular" region exists
+exec-sql
+ALTER DATABASE d ADD REGION "us-east-1";
+----
+pq: cannot add region "us-east-1" to database d
+HINT: you must add a PRIMARY REGION first using ALTER DATABASE d PRIMARY REGION "us-east-1"
+
+# we also can't add a primary region to our regionless database out-of-the-box now...
+exec-sql
+ALTER DATABASE d PRIMARY REGION "us-east-1";
+----
+pq: type "d.public.crdb_internal_region" already exists
+DETAIL: multi-region databases employ an internal enum called crdb_internal_region to manage regions which conflicts with the existing object
+HINT: object "crdb_internal_regions" must be renamed or dropped before adding the primary region
+
+# ...let's listen to the error message
+exec-sql
+DROP TYPE d.public.crdb_internal_region;
+----
+
+exec-sql
+ALTER DATABASE d PRIMARY REGION "us-east-1";
+----
+
+query-sql
+SELECT region FROM [SHOW REGIONS FROM DATABASE d] ORDER BY 1;
+----
+us-east-1
+
+query-sql
+SHOW TABLES FROM d;
+----
+public t table root <nil> REGIONAL BY TABLE IN PRIMARY REGION
+
+# ensure that tables belonging to d can be modified & have the correct values
+exec-sql
+INSERT INTO d.t VALUES (6), (7);
+----
+
+query-sql
+SELECT * FROM d.t;
+----
+1
+2
+3
+4
+5
+6
+7
+
+subtest end
+
+subtest end
+
+# create cluster with a single region and test table restore with strip_localities
+subtest restore_regionless_on_single_region_table
+new-cluster name=s4 share-io-dir=s1 allow-implicit-access localities=us-east-1
```

Comment:
ditto to message above: i'd avoid cluster creation, run `DROP database d` and proceed with the test.